File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> freq(nums.begin(), nums.end());
        int max = 0;
        vector<int> starts;
        for (int i = 0; i < nums.size(); i++) {
            if (!freq.count(nums[i] - 1)) {
                starts.push_back(nums[i]);
            }
        }
        for (int i = 0; i < starts.size(); i++) {
            int dist = 1;
            for (int j = 1; j < nums.size(); j++) {
                if (freq.count(starts[i] + j)) {
                    dist++;
                }
                else {
                    break;
                }
            }
            if (dist > max) {
                max = dist;
            }
        }
        return max;
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-2.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        starts = []
        ends = []
        for interval in intervals:
            starts.append(interval.start)
            ends.append(interval.end)

        count = 0
        res = 0
        starts.sort()
        ends.sort()
        s = 0
        e = 0
        while s < len(starts):
            if starts[s] < ends[e]:
                s += 1
                count += 1
                res = max(count, res)
            else:
                e += 1
                count -= 1
        return res